tests/test_time_series_graph.py

```diff
@@ -273,6 +273,22 @@ def test_replace_node(self):
         # check 'X1 lag(n=1)' does not exist
         self.assertFalse(ts_cg.node_exists('X1 lag(n=1)'))
 
+    def test_replace_edge(self):
```

Comment:
Can you add an error case where you try to replace with a new edge that flows backwards in time? Just confirm it errors.